### ISSUE TITLE
Using Kafka Streams Branching causes NPE

### ISSUE THREAD

**Issue by GFriedrich** [spring-cloud/spring-cloud-stream#2820]
**Description**
When using Kafka Streams branching (so returning an array of `KStream` objects from a function), it will cause an NPE when setting the functions via the `spring.cloud.function.definition`, which is required when having more than one functional bean on the Spring context.

**Steps to reproduce the behavior**
1. Use the Kafka Streams branching example from https://github.com/spring-cloud/spring-cloud-stream-samples/tree/main/kafka-streams-samples/kafka-streams-branching
2. Define the function via `spring.cloud.function.definition`
3. When running the example, see that an NPE is being triggered like:
```
Caused by: java.lang.NullPointerException: Cannot invoke "Object.toString()" because "functionType" is null
	at org.springframework.cloud.stream.function.FunctionConfiguration$FunctionBindingRegistrar.filterEligibleFunctionDefinitions(FunctionConfiguration.java:980)
	at org.springframework.cloud.stream.function.FunctionConfiguration$FunctionBindingRegistrar.afterPropertiesSet(FunctionConfiguration.java:831)
```

**Suggested Fix**
I've patched a local copy at the code path of
https://github.com/spring-cloud/spring-cloud-stream/blob/ca2d3046b9bed5bb29b82a55044fe8fa809a760d/core/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/FunctionConfiguration.java#L1027-L1032
with the following
```java
Object functionBean = this.applicationContext.getBean(functionName);
Type functionType = FunctionTypeUtils.discoverFunctionType(functionBean, functionName, (GenericApplicationContext) this.applicationContext);
if (functionType == null) {
  eligibleDefinition = false;
} else {
  String functionTypeStringValue = functionType.toString();
  if (functionTypeStringValue.contains("KTable") || functionTypeStringValue.contains("KStream")) {
    eligibleDefinition = false;
  }
}
```

**Version of the framework**
4.0.3

Let me know if you've any further questions.

**Comment by sobychacko** [comment]
@GFriedrich Thanks for finding this issue. It is now addressed on both `main` and the `4.0.x` branches.

**Comment by GFriedrich** [comment]
Also thanks for this fix. Much appreciated.
